src/Interpreters/AddDefaultDatabaseVisitor.h

```diff
@@ -138,13 +154,17 @@ class AddDefaultDatabaseVisitor
 
     void visit(const ASTTableIdentifier & identifier, ASTPtr & ast) const
     {
-        if (!identifier.compound())
-        {
-            auto qualified_identifier = std::make_shared<ASTTableIdentifier>(database_name, identifier.name());
-            if (!identifier.alias.empty())
-                qualified_identifier->setAlias(identifier.alias);
-            ast = qualified_identifier;
-        }
+        /// Already has database.
+        if (identifier.compound())
+            return;
+        /// There is temporary table with such name, should not be rewritten.
+        if (external_tables.count(identifier.shortName()))
```

Comment:
```suggestion
        if (external_tables.contains(identifier.shortName()))
```

Reply:
Indeed, looks better, the reason I prefer to use `count` over `contains` is that before I have to remember about arcadia built that has pretty limited c++20 support, and `contains` wasn't supported, but anyway this is not relevant anymore.